docs/decisions/0004-axios-caching-implementation.rst

```diff
@@ -0,0 +1,87 @@
+
+Each request can also override the default cache configurations. `The How To document about caching <https://github.com/edx/frontend-platform/blob/master/docs/how_tos/caching.rst>`_ has more detailed examples about how this caching layer can be used.
+
+
+Implementation Details
+~~~~~~~~~~~~~~~~~~~~~~
+
+For both versions of the current AxiosJwtAuthService HTTP clients, the authenticated client and the base unauthenticated clients, there are now two additional versions that use the cache (i.e. CachedAuthenticated and CachedUnauthenticated). Bringing the total number of httpClient instances up to four.
+In order to keep the interface more simple the cached clients don’t have their own getter functions exposing this detail, in order to use the cached http client micro frontends pass in a param for using the cache with the current getter functions.
+
+In the case of the cached clients failing to get created the uncached clients will be returned and used instead.
+
+The async hydrateAuthenticatedUser function will continue only ever using the uncached authenticated HTTP client so there won't be any concerns about the cache changing expected behavior of authentication.
+
+Only the AxiosJwtAuthService currently has cached http clients.
+
+Decisions
+---------
+
+By using the existing axios-cache-adapter we were able to avoid needing to implement our own caching strategy and forgo having micro frontends need to implement their own front end caching strategies.
+
+The default max-age for cached data was decided to be 5 minutes for now. The decision was based on it being a decent amount of the storage of cached data even for authentication specific requests. The assumption is made that sers won’t be frequently switching to different accounts on their browser frequently enough to have a wide impact and if they do then they will only see cached data for a maximum of 5 minutes. Micro frontends are also able to override the cache time to suit their own needs.
+The default max-age can also be changed at a later time if the current default of 5 minutes is deemed to be inadequate.
+
+Tests written to verify the cache adapter integrates well with the current interceptors make real external API calls to https://httpbin.org/ as per the tests in the axios-cache-adapter source code.
+It was found to be impossible to mock out the exact spot the cache client makes an outside request without also mocking over the cache clients caching functionality. This wasn’t the preferred approach to testing because it makes external network requests but since it is the approach that was used in the axios-cache-adapter tests it was decided to be fine.
+
+Alternatives
+------------
+
+We discussed the pros and cons of having a front-end platform cache implementation for micro front ends to use versus micro front ends implementing their own front end caching strategies utilizing Local/Session storage, ETags, and endpoints utilizing Cache-Control headers.
```

Comment:
Why did we not embrace any of these alternatives?  Or could these be used in conjunction with the implementation we settled on?

Reply:
They could be used in conjunction if each MFE decides they want to implement them as well. The main idea was that having this caching strategy be in frontend-platform makes it widely available and accessible to MFEs instead of deciding that each MFE needs to implement their own solution/reimplement another solution copied from somewhere else

Reply:
Cool, could you add some version of those two sentences to this section? :)